Common/Retranslator/src/Retranslator.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Retranslator.hpp"
+#include <typeinfo>
+
+using namespace AST;
+
+std::unique_ptr<Retranslator> Retranslator::instance = nullptr;
+
+ASTTree Retranslator::parseTree(std::map<std::string, Shape*> map){
+    
+    Node* root = new Node("root");
+    root->addChild(new Node("STARTGRAPH"));
+
+    for (const auto& elem : map)
+    {
+        root->addChild(new Node("STATEMENT"));
+        // if (dynamic_cast<Note*>(elem.second))
+        // {
+        //     root->addChild(makeNote(elem));
+        // }
+        if (dynamic_cast<Line*>(elem.second))
+        {
+            root->addChild(makeLink(elem));
+        }
+        else 
+        {
+            root->addChild(makeObject(elem));
+        }
+    }
+
+    root->addChild(new Node("ENDGRAPH"));
+
+    return ASTTree(root);
+}
+
+Node* doubleNode(std::string key, std::string value, bool isText){
+    Node* property = new Node("property");
+    property->addChild(new Node("PROPERTY_KEY"));
+    property->childNodes[0]->addChild(new Node(key));
+    property->addChild(new Node("="));
+    if (isText)
+        property->addChild(new Node("TEXT"));
+    else 
+        property->addChild(new Node("NUMBER"));
+    property->childNodes[2]->addChild(new Node(value));
+
+    return property;
+}
+
+void addProperty(Node* node, std::string key, std::string value, bool isText = false)
+{
+    node->addChild(doubleNode(key, value, isText));
+    node->addChild(new Node(";"));
+}
+
+Node* Retranslator::makeObject(std::pair<std::string, Shape*> pair) const
+{
+    Shape* shape = pair.second;
+    Node* node = new Node("object_decl");
+    node->addChild(new Node("SHAPE"));
+
+    if (dynamic_cast<Circle*>(shape))
+        node->childNodes[0]->addChild(new Node("circle"));
+    else if (dynamic_cast<Diamond*>(shape))
+        node->childNodes[0]->addChild(new Node("diamond"));
+    else if (dynamic_cast<Reactangle*>(shape))
+        node->childNodes[0]->addChild(new Node("reactangle"));
+
+    node->addChild(new Node("ID"));
+    node->childNodes[1]->addChild(new Node(pair.first));
+
+    node->addChild(new Node("{"));
+
+    //общая часть
+    
+    addProperty(node, "text", shape->text, true);
+    addProperty(node, "сolor", std::to_string(static_cast<short>(shape->style.color)));
```

Comment:
```suggestion
    addProperty(node, "сolor", std::to_string(static_cast<short>(shape->style.color)), true);
```
Ведь color тоже текстом задаётся у нас